### ISSUE TITLE
Crash - Pressing home screen when fragment is open

### ISSUE THREAD

**Issue by silencecm** [silencecm/AndroidBAC#21]
Pressing the home screen when a fragment (have drink, edit, information) is open causes the application to crash. Stacktrace:
```
11-16 22:41:05.525  26848-26848/com.silencecm.BAC E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.silencecm.BAC, PID: 26848
    java.lang.RuntimeException: Parcelable encountered IOException writing serializable object (name = com.silencecm.BAC.model.BACUser)
            at android.os.Parcel.writeSerializable(Parcel.java:1388)
            at android.os.Parcel.writeValue(Parcel.java:1335)
            at android.os.Parcel.writeArrayMapInternal(Parcel.java:638)
            at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1313)
            at android.os.Bundle.writeToParcel(Bundle.java:1096)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.app.FragmentState.writeToParcel(Fragment.java:146)
            at android.os.Parcel.writeTypedArray(Parcel.java:1191)
            at android.app.FragmentManagerState.writeToParcel(FragmentManager.java:380)
            at android.os.Parcel.writeParcelable(Parcel.java:1357)
            at android.os.Parcel.writeValue(Parcel.java:1262)
            at android.os.Parcel.writeArrayMapInternal(Parcel.java:638)
            at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1313)
            at android.os.Bundle.writeToParcel(Bundle.java:1096)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.app.ActivityManagerProxy.activityStopped(ActivityManagerNative.java:3080)
            at android.app.ActivityThread$StopInfo.run(ActivityThread.java:3772)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5942)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
     Caused by: java.io.NotSerializableException: com.silencecm.BAC.model.BACTime
            at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1344)
            at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
            at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
            at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
            at java.io.ObjectOutputStream.writeFieldValues(ObjectOutputStream.java:959)
            at java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:360)
            at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1054)
            at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
            at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
            at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
            at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
            at android.os.Parcel.writeSerializable(Parcel.java:1383)
            at android.os.Parcel.writeValue(Parcel.java:1335)
            at android.os.Parcel.writeArrayMapInternal(Parcel.java:638)
            at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1313)
            at android.os.Bundle.writeToParcel(Bundle.java:1096)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.app.FragmentState.writeToParcel(Fragment.java:146)
            at android.os.Parcel.writeTypedArray(Parcel.java:1191)
            at android.app.FragmentManagerState.writeToParcel(FragmentManager.java:380)
            at android.os.Parcel.writeParcelable(Parcel.java:1357)
            at android.os.Parcel.writeValue(Parcel.java:1262)
            at android.os.Parcel.writeArrayMapInternal(Parcel.java:638)
            at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1313)
            at android.os.Bundle.writeToParcel(Bundle.java:1096)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.app.ActivityManagerProxy.activityStopped(ActivityManagerNative.java:3080)
            at android.app.ActivityThread$StopInfo.run(ActivityThread.java:3772)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:145)
            at android.app.ActivityThread.main(ActivityThread.java:5942)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

**Comment by silencecm** [comment]
Fixed by adding onPause() lifecycle events to all fragments and calling dismiss() on the active dialog fragment.

**Comment by silencecm** [comment]
I also experienced this when working on #49 same stacktrace. I implemented the serializable method `writeObject()` and added `bacTime = null` to resolve the issue. The above `dismiss()` fix won't work because no longer using dialog fragment. I'm guessing a strong reference to `BACUser` is passed to the fragment during `putSerializable()` which in turn is trying to serialize `BACUser`. I think nulling `BACTime` should fix this completely, making the BACUser safe to be serializable.
